jsapp/js/searches.es6

```diff
@@ -119,21 +119,24 @@ function SearchContext(opts={}) {
     },
     // remove asset from all search store lists
     removeAsset(assetOrUid, isNonOwner) {
+      let asset;
+      if (typeof assetOrUid === 'object') {
+        asset = assetOrUid;
+      } else {
+        asset = stores.selectedAsset.asset || stores.allAssets.byUid[assetOrUid];
+      }
+      // non-owner self permission removal only gives an assetUid string, not
+      // an object; for consistency we make it an object here
+      if (!asset) {
+       asset = {uid: assetOrUid};
+      }
+
       // only update things if given asset matches the current context types or
       // a non-owner removed their own permissions
-      if (isNonOwner || this.state.defaultQueryFilterParams?.assetType.includes(asset.assetType)) {
-        let asset;
-        if (typeof assetOrUid === 'object') {
-          asset = assetOrUid;
-        } else {
-          asset = stores.selectedAsset.asset || stores.allAssets.byUid[assetOrUid];
-        }
-
-        // non-owner self permission removal only gives an assetUid string, not
-        // an object; for consistency we make it an object here
-        if (!asset) {
-         asset = {uid: assetOrUid};
-        }
+      if (
+        isNonOwner ||
+        this.state.defaultQueryFilterParams?.assetType.includes(asset.assetType)
```

Comment:
Do you think we need the fake asset object to also have the `assetType` so this will go through?